pkg/assets/create/creater.go

```diff
@@ -57,45 +55,50 @@ func EnsureManifestsCreated(ctx context.Context, manifestDir string, restConfig
 		options.StdErr = os.Stderr
 	}
 
-	// Default QPS in client (when not specified) is 5 requests/per second
-	// This specifies the interval between "create-all-resources", no need to make this configurable.
-	interval := 200 * time.Millisecond
-
-	// Retry creation until no errors are returned or the timeout is hit.
 	var (
 		lastCreateError      error
-		retryCount           int
 		mapper               meta.RESTMapper
 		needDiscoveryRefresh bool = true
 	)
-	err = wait.PollImmediateUntil(interval, func() (bool, error) {
-		retryCount++
+	for retryCount := 1; ; retryCount++ {
+		if retryCount > 1 {
+			select {
+			case <-ctx.Done():
+				if lastCreateError == nil {
+					return ctx.Err()
+				}
+				// Return the last observed set of errors from the create process instead of timeout error.
+				return lastCreateError
+			default:
+			}
+		}
 		// If we get rest mapper error, we need to pull updated discovery info from API server
 		if needDiscoveryRefresh {
 			mapper, err = fetchLatestDiscoveryInfoFn(dc)
 			if err != nil {
 				if options.Verbose {
 					fmt.Fprintf(options.StdErr, "[#%d] failed to fetch discovery: %s\n", retryCount, err)
 				}
-				return false, nil
+				lastCreateError = err
+				continue
 			}
 		}
-		lastCreateError, needDiscoveryRefresh = create(manifests, client, mapper, options)
-		if lastCreateError == nil {
-			return true, nil
+		err, needDiscoveryRefresh = create(ctx, manifests, client, mapper, options)
+		if err == nil {
+			return nil
+		}
+		if err == context.Canceled || err == context.DeadlineExceeded {
```

Comment:
i believe this is handled in `context.Done()` channel already.

Reply:
> i believe this is handled in `context.Done()` channel already.

Yeah, this appears to just duplicate the existing code.